tests/integ/modin/test_merge_asof.py

```diff
@@ -0,0 +1,257 @@
+    left_right_native_df_switch_column_order, allow_exact_matches, direction
+):
+    left_native_df, right_native_df = left_right_native_df_switch_column_order
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    native_output = native_pd.merge_asof(
+        left_native_df,
+        right_native_df,
+        on="a",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    snow_output = pd.merge_asof(
+        left_snow_df,
+        right_snow_df,
+        on="a",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    eval_snowpark_pandas_result(snow_output, native_output, lambda df: df)
+
+
+@allow_exact_matches
+@direction
+@sql_count_checker(query_count=1, join_count=1)
+def test_merge_asof_left_right_on(
+    left_right_native_df_left_right_by, allow_exact_matches, direction
+):
+    left_native_df, right_native_df = left_right_native_df_left_right_by
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    native_output = native_pd.merge_asof(
+        left_native_df,
+        right_native_df,
+        left_on="a",
+        right_on="b",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    snow_output = pd.merge_asof(
+        left_snow_df,
+        right_snow_df,
+        left_on="a",
+        right_on="b",
+        allow_exact_matches=allow_exact_matches,
+        direction=direction,
+    )
+    eval_snowpark_pandas_result(snow_output, native_output, lambda df: df)
+
+
+@sql_count_checker(query_count=1, join_count=1)
+def test_merge_asof_timestamps(left_right_timestamp_data):
+    left_native_df, right_native_df = left_right_timestamp_data
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    native_output = native_pd.merge_asof(left_native_df, right_native_df, on="time")
+    snow_output = pd.merge_asof(left_snow_df, right_snow_df, on="time")
+    eval_snowpark_pandas_result(snow_output, native_output, lambda df: df)
+
+
+@sql_count_checker(query_count=0)
+def test_merge_asof_nearest_unsupported(left_right_timestamp_data):
+    left_native_df, right_native_df = left_right_timestamp_data
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    with pytest.raises(
+        NotImplementedError,
+        match="Snowpark pandas merge_asof method only supports directions 'forward' and 'backward'",
+    ):
+        pd.merge_asof(left_snow_df, right_snow_df, on="time", direction="nearest")
+
+
+@sql_count_checker(query_count=0)
+def test_merge_asof_params_unsupported(left_right_timestamp_data):
+    left_native_df, right_native_df = left_right_timestamp_data
+    left_snow_df, right_snow_df = pd.DataFrame(left_native_df), pd.DataFrame(
+        right_native_df
+    )
+    with pytest.raises(
```

Comment:
shouldn't we have a separate pytest.raises for each one, as otherwise the first one will trip the raises?

Reply:
Good catch, done.